packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,413 @@
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      // tab to focus first bumper
       cy.realPress('Tab');
-      cy.focused().should('have.id', 'a');
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('focuses first focusable element when focusing outside of FTZ with 0 tabbable items', () => {
+      setProps({});
+
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      // try to focus on button outside FTZ
+      cy.get('#before').focus();
+      // it focuses first button inside FTZ instead
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('focuses previously focused element when focusing outside of FTZ with 0 tabbable items', () => {
+      setProps({ focusPreviouslyFocusedInnerElement: true });
+
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      // try to focus on button outside FTZ
+      cy.get('#before').focus();
+      // it focuses last focused button inside FTZ instead
+      cy.focused().should('have.id', 'mid');
+    });
+  });
+
+  describe('Imperatively focusing the FTZ', () => {
+    function imperativeFocus() {
+      cy.window().then(win => (win as FTZTestWindow).imperativeFocus!());
+    }
+
+    beforeEach(() => {
+      cy.loadStory(ftzStoriesTitle, 'ImperativeFocus');
+    });
+
+    it('goes to previously focused element when focusing the FTZ', async () => {
+      setProps({ focusPreviouslyFocusedInnerElement: true });
+
+      // Manually focusing FTZ when FTZ has never had focus within should go to 1st focusable inner element.
+      imperativeFocus();
+      cy.focused().should('have.id', 'first');
+
+      // Focus inside the trap zone, not the first element.
+      cy.get('#last').focus();
+      cy.focused().should('have.id', 'last');
+
+      // Focus outside the trap zone
+      cy.get('#after').focus();
+      cy.focused().should('have.id', 'after');
+
+      // Manually focusing FTZ should return to originally focused inner element.
+      imperativeFocus();
+
+      cy.focused().should('have.id', 'last');
+    });
+
+    it('goes to first focusable element when focusing the FTZ', async () => {
+      setProps({ focusPreviouslyFocusedInnerElement: false });
+
+      // Manually focusing FTZ when FTZ has never had focus within should go to 1st focusable inner element.
+      imperativeFocus();
+      cy.focused().should('have.id', 'first');
+
+      // Focus inside the trap zone, not the first element.
+      cy.get('#last').focus();
+      cy.focused().should('have.id', 'last');
+
+      // Focus outside the trap zone
+      cy.get('#after').focus();
+      cy.focused().should('have.id', 'after');
+
+      // Manually focusing FTZ should go to the first focusable element.
+      imperativeFocus();
+      cy.focused().should('have.id', 'first');
+    });
+  });
+
+  it('maintains a proper stack of FocusTrapZones as more are mounted/unmounted', () => {
```

Comment:
this test is insane, and probably makes sense here to  use unit testing, browser testing shouldn're really test mount/unmount

Reply:
Agreed, but due to all the mocking required, I'm not convinced that the unit test was actually covering what it claimed to cover.

Reply:
Should seriously consider what are the interaction implications of testing nested FTZs and test that instead of a stack being registered correctly on mount/unmount

Reply:
I changed it to click a bunch of buttons to test the actual interaction.